phpBB/includes/functions_messenger.php

```diff
@@ -1853,53 +1853,16 @@ function digest_md5($username, $password)
 */
 function mail_encode($str, $eol = "\r\n")
 {
-	// Check if string contains ASCII only characters
-	$is_ascii = strlen($str) === utf8_strlen($str);
-
-	// Define start delimimter, end delimiter and spacer
-	// Use the Quoted-Printable encoding for ASCII strings to avoid unnecessary encoding in Base64
-	$start = $is_ascii ? "=?US-ASCII?Q?" : "=?UTF-8?B?";
-	$end = "?=";
-	$delimiter = "$eol ";
-
-	// Maximum encoded-word length is 75 as per RFC 2047 section 2.
-	// $split_length *must* be a multiple of 4, but <= 75 - strlen($start . $delimiter . $end)!!!
-	$split_length = 75 - strlen($start . $delimiter . $end);
-	$split_length = $split_length - $split_length % 4;
-
-	// Use the Quoted-Printable encoding for ASCII strings to avoid unnecessary encoding in Base64
-	// quoted_printable_encode() splits lines at length of 75 characters with =\r\n delimiter, amend this feature
-	$encoded_str = $is_ascii ? str_replace("=\r\n", '', quoted_printable_encode($str)) : base64_encode($str);
-
-	// If encoded string meets the limits, we just return with the correct data.
-	if (strlen($encoded_str) <= $split_length)
-	{
-		return $start . $encoded_str . $end;
-	}
-
-	// If there is only ASCII data, we just return what we want, correctly splitting the lines.
-	if ($is_ascii)
-	{
-		return $start . implode($end . $delimiter . $start, str_split($encoded_str, $split_length)) . $end;
-	}
+	// Correctly decode html entities
+	$str = html_entity_decode($str);
```

Comment:
Looks like something new awaiting to be approved.

https://github.com/phpbb/phpbb/pull/6320/files#diff-def4c3d80a12a0d27ddce789bcc0809ead44c8e488e2f05181d8846e1a0e5324R1866-R1867